### DISCUSSION TITLE
Installing modern-cv locally (on MacOS)

### DISCUSSION THREAD

**Discussion by mgrimace** [DeveloperPaul123/modern-cv#61] [General]
I'm going to post my step-by-step novice instructions here in case I forget, as it was a bit harder for me to figure out how to install this initially since I'd never used Typst before. For windows you can probably just use the bundled installer script but I couldn't figure that out on Mac.

# Install modern-cv locally on a mac
## Install typst and fonts
- In terminal `brew install typst`
- Install the fonts as listed in the modern-cv readme manually by downloading them, and importing them into Font Book
## Install modern-cv
- download the source code.zip found on github in the releases section. Select the most current release, which in this example is 0.6.0. Unzip/extract it.
- there should now be a folder in your 'Downloads' folder called `modern-cv-0.6.0`
- in Finder, go to `~/Library/Application Support` (e.g., open finder, select 'go' in the menubar, 'go to folder' and paste that in)
- create a new folder here called `typst`, open that folder and create a new folder called `packages`, open that and create a new folder called `local`
- inside `local`, create a new folder called `modern-cv`, and finally, inside there, create one last folder named with the release number that you downloaded, for example `0.6.0`.
- you should have a pathway now that looks like: `/Users/[your user name]/Library/Application Support/typst/packages/local/modern-cv/0.6.0`
- copy/paste or drag the contents of `modern-cv-0.6.0` that you downloaded and extracted earlier *into* this new `0.6.0` folder. 
## Customize the template
- You can make changes to the lib.typ in this folder, for example adding/removing sections.
## Use your new customized template
- In your resume.typ or cover.typ, change the first line to `#import "@local/modern-cv:0.6.0": *` by using '@local' it tells it to use the template that we set up locally. 
- Optionally, add new tag folders as they are released (e.g., modern-cv/0.7.0, etc.)

I'm sure there's easier and more efficient was of doing this, and hopefully this helps.

**Comment by ptsouchlos** [comment]
Thanks for this. I think I do need to create some sort of "build" instruction for different platforms and maybe add more scripts to help this.

**Comment by ptsouchlos** [comment]
There's now new scripts you can use to easily install `typst` locally though I haven't tested them on macOS (only) linux.
